File: 03-frontend-booking-system/booking-system-web/src/app/features/slots/dtos/slot-request-dto.ts
import { TimeRange } from "@shared/model/time-range";

export class SlotRequestDto {
    eventId!: number;
    slotName!: string;
    slotDescription?: string;
    slotStartTime!: string; //ISO string
    slotEndTime!: string; //ISO string
    maxBookPerInterval?: number;
    slotIntervalMinutes?: number;
    slotFrequencyIntervalMinutes?: number;
    businessDaysHours?: Record<number, TimeRange[]>;
    businessTimeZone?: string;
    businessAllowOt?: boolean;
    flexibleDaysHours?: TimeRange[];
}

File: 02-backend-booking-system/src/main/java/com/littlek4za/booking_system/dtos/SlotRequestDto.java
package com.littlek4za.booking_system.dtos;

import java.time.Instant;
import java.util.List;
import java.util.Map;

import com.littlek4za.booking_system.models.InstantRange;
import com.littlek4za.booking_system.models.TimeRange;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

public record SlotRequestDto(

    @NotNull(message = "is required")
    Long eventId,

    @NotBlank(message = "is required")
    @Size(min = 1, max = 350, message = "Slot Name must be between 1 and 350 characters")
    String slotName,

    @Size(max = 2500, message = "Slot Description must not exceed 2500 characters")
    String slotDescription,

    Instant slotStartTime,

    Instant slotEndTime,
    
    Integer maxBook,

    Integer slotIntervalMinutes,

    Integer slotFrequencyIntervalMinutes,

    Map<Integer, List<TimeRange>> businessDaysHours,

    List<InstantRange> flexibleDaysHours

) {} 
